Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -190,42 +191,53 @@ struct KlaviyoState: Equatable, Codable {
         var attributes = profile.data.attributes
         var location = profile.data.attributes.location ?? .init()
         var properties = profile.data.attributes.properties.value as? [String: Any] ?? [:]
+        let updatedProfile = Profile.updateProfileWithProperties(dict: pendingProfile)
 
-        // Optionally (if not already specified) overwrite attributes and location with pending profile information.
-        for (key, value) in pendingProfile {
-            switch key {
-            case .firstName:
-                attributes.firstName = attributes.firstName ?? value.value as? String
-            case .lastName:
-                attributes.lastName = attributes.lastName ?? value.value as? String
-            case .address1:
-                location.address1 = location.address1 ?? value.value as? String
-            case .address2:
-                location.address2 = location.address2 ?? value.value as? String
-            case .title:
-                attributes.title = attributes.title ?? value.value as? String
-            case .organization:
-                attributes.organization = attributes.organization ?? value.value as? String
-            case .city:
-                location.city = location.city ?? value.value as? String
-            case .region:
-                location.region = location.region ?? value.value as? String
-            case .country:
-                location.country = location.country ?? value.value as? String
-            case .zip:
-                location.zip = location.zip ?? value.value as? String
-            case .image:
-                attributes.image = attributes.image ?? value.value as? String
-            case .latitude:
-                location.latitude = location.latitude ?? value.value as? Double
-            case .longitude:
-                location.longitude = location.longitude ?? value.value as? Double
-            case let .custom(customKey: customKey):
-                properties[customKey] = properties[customKey] ?? value.value
-            }
+        if let firstName = updatedProfile.firstName {
```

Comment:
@ndurell as you identified this logic is so that if any stuff was set in the attributes of a profile they are preserved.